protected/components/CustomGridView.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * CustomGridView Class
+ *
+ * Extends CGridView to add custom pagination styles by using bootstrap classes.
+ *
+ * Example usage:
+ * ```php
+ * $this->widget('CustomGridView', array(
+ *   'dataProvider' => $dataProvider,
+ *   // other options here...
+ * ));
+ * ```
+ *
+ * Note: This class assumes jQuery is loaded. It uses AJAX to refresh the pagination state.
+ */
+
+
+Yii::import('zii.widgets.grid.CGridView');
+
+class CustomGridView extends CGridView {
+
+    public function init() {
+        $this->pager = array_merge(
+            $this->pager,
+            array(
+                'header' => '',
+                'htmlOptions' => array('class' => 'pagination'),
+            )
+        );
+
+        $this->pagerCssClass = 'pagination-container';
+
+        parent::init();
+
+        Yii::app()->clientScript->registerScript('pagination-adjustment', '
```

Comment:
this jQuery script adds rounded borders to first and last visible elements, attempting this with CSS-only was impractical